src/CacheFactory.php

```diff
@@ -67,7 +67,7 @@ function ($provider) use ($container) {
                 break;
 
             default:
-                $cache = new $config['class']();
+                $cache = $container->has($config['class']) ? $container->get($config['class']) : new $config['class']();
```

Comment:
I like this, good call.

@DASPRiD I think this could be considered a "low risk" BC break (if someone has defined the value of `$config['class']` in their container already, this may cause unintended consequences); this should probably be documented, maybe in the `README.md`, and pitch it as a "feature", wdyt? :grin:

Reply:
i don't think its a problem, if the value of `$config['class']` is a class ( which it supposed to be) then the container is supposed to return an instance of that class or if its an interface, its gonna return a class that implement it.

Reply:
That's a common convention, yes, but there's no guarantee someone isn't consuming their container differently. The point is, assumptions would change with this. The ball is in @DASPRiD 's court for this tho :)

Reply:
@DASPRiD what do you think about this ?

Reply:
I don't think there there is a problem with this; The built-in caches are configured via their class names, so it's unlikely that someone has registered a service with the same key. I'd call that a low-risk BC break, which most likely doesn't affect anyone.

Reply:
good 👍